### DISCUSSION TITLE
So....exactly how does a person use this?

### DISCUSSION THREAD

**Discussion by sgjohnson1981** [freeplane/gradle-plugin#6] [Q&A]
I'm not an experienced developer, so after staring at the instructions for awhile I'm still not sure how to use this. I actually needed the idea to pop into my head that what I was looking at was something more than a build.gradle suggestion. The clue was "Plugin" was capitalized, so maybe that's a Gradle thing? After reading [this](https://docs.gradle.org/current/userguide/plugins.html) I finally realized it's a script-type plugin. But I still can't find anywhere telling me what to do with it. Where does it go in relation to the freeplane source and/or your add-on source? Does it need to be built using build.gradle in the root of the plugin's source? What happens when it's built? Where do you put the results of the build?

Also, what does this sentence mean: "Add your dependencies either to special configuration addon to be included in the created add-on installer."
What is a "special configuration addon"? Why is there an "either" there? I only see one thing.

**Comment by EdoFro** [comment]
hi @sgjohnson1981,
I have learned a little more since my last approach to this plugin. I will try it again and report back if I can use it. Hopefully this weekend.

BR,

edo

**Comment by macmarrum** [comment]
@sgjohnson1981 
My suggestion is that you start by creating a Freeplane add-on the old-fashioned way, without this gradle-plugin. Once you grasp the concepts of what it means to create an add-on for Freeplane, you'll be in a better position to understand the reason for this project and what benefits it can bring for you.

**Reply by macmarrum** [reply]
Here's an example of a Freeplane add-on: https://github.com/macmarrum/freeplane-MapObfuscator/releases

**Reply by sgjohnson1981** [reply]
@macmarrum Already done that. I messed around with @EdoFro's groovy console to change the colors to a dark mode.

**Reply by macmarrum** [reply]
OK, great :+1: 
You have probably followed the steps from [README](https://github.com/freeplane/gradle-plugin/blob/master/README.md).
It might be beneficial if you could describe what went well and at which step you got stuck, so that you can get help and the instructions can be improved :smile:

**Reply by sgjohnson1981** [reply]
@macmarrum I was hoping someone could just answer my questions in the OP, for now at least. For me, learning is just as much doing as it is understanding what's going on underneath what I'm doing.

**Comment by dpolivaev** [comment]
This plugin is a Gradle plugin that can be used to simplify Freeplane add-on development using Gradle. The code contains an example showing how it can be used.

https://github.com/freeplane/gradle-plugin/tree/master/examples/greetings
